src/integrations/stripe.js

```diff
@@ -0,0 +1,101 @@
+import { loadStripe } from '@stripe/stripe-js';
+
+const stripePromise = new Promise((resolve) => {
+  document.addEventListener('stripe_key_loaded', (e) => {
+    const promise = loadStripe(e.detail.key);
+    promise.then((stripe) => {
+      resolve(stripe);
+    });
+  });
+});
+
+export function getStripePromise() {
+  return stripePromise;
+}
+export async function getStripe() {
+  return await getStripePromise();
+}
+
+export function installStripe(stripeConfig) {
+  // eslint-disable-next-line camelcase
+  if (stripeConfig?.metadata?.client_key) {
+    document.dispatchEvent(
+      new CustomEvent('stripe_key_loaded', {
+        detail: { key: stripeConfig.metadata.client_key }
+      })
+    );
+  }
+}
+
+export function getStripeCustomerFieldValues(stripeConfig, fieldValues) {
+  if (!stripeConfig?.metadata?.customer_field_mappings) return {};
+  const customer = getObjectMappingValues(
+    stripeConfig.metadata.customer_field_mappings,
+    fieldValues
+  );
+  // the shipping object in customer must have a name property (stripe requires this),
+  //  but we don't want to make the mapping required.  So default it here.
+  customer.shipping &&
+    !customer.shipping.name &&
+    (customer.shipping.name = 'Name not supplied');
+  return customer;
+}
+function getObjectMappingValues(mappingObj, fieldValues) {
+  return Object.entries(mappingObj).reduce((result, [key, mappingInfo]) => {
+    if (mappingInfo.id)
+      result[key] =
+        fieldValues[
+          mappingInfo.type === 'servar' && mappingInfo.servar
+            ? mappingInfo.servar
+            : mappingInfo.id
+        ];
+    // sub-object
+    else result[key] = getObjectMappingValues(mappingInfo, fieldValues);
+    return result;
+  }, {});
+}
+
+export function getPaymentMethodFieldsOnStep(step) {
+  return step.servar_fields.filter(
+    (field) => field.servar.type === 'payment_method'
+  );
+}
+/**
+ * Sets up a payment method (i.e. a card) in stripe under a customer.
+ * @param {*} client
+ * @param {*} formFieldValues the form's field values
+ * @param {*} paymentMethodFieldId the payment method field id
+ * @param {*} stripeConfig the integration config for stripe.  Non-null.
+ * @param {*} stripe the stripe api object.
+ * @param {*} cardElement tthe stripe cardElement (from stripe Elements).
+ */
+export async function setupPaymentMethod(
+  client,
+  formFieldValues,
+  paymentMethodFieldId,
+  stripeConfig,
+  stripe,
+  cardElement
+) {
+  if (!stripeConfig || Object.keys(stripeConfig).length === 0) return null;
+
+  const { intent_secret: intentSecret } = await client.setupPaymentIntent(
```

Comment:
can any of this payment method setup logic be moved to the backend? in general whatever we can handle on the backend, we should. it'll also keep the react lib more lightweight

Reply:
To the best of my knowledge we have moved everything possible to the backend.  The stripe.confirmCardSetup must be done from the client api in order to send the cardElement with all the card info like number, exp. cvc, etc.